BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -1223,9 +1231,9 @@ public void CheckDetonationState()
                                         //We found a hit a different vessel than ours
                                         if (DetonateAtMinimumDistance)
                                         {
-                                            var distance = Vector3.Distance(partHit.transform.position, vessel.CoM);
-                                            var predictedDistance = Vector3.Distance(AIUtils.PredictPosition(partHit.transform.position, partHit.vessel.Velocity(), partHit.vessel.acceleration, Time.deltaTime), AIUtils.PredictPosition(vessel, Time.deltaTime));
-                                            if (distance > predictedDistance && distance > Time.fixedDeltaTime * (float)vessel.srfSpeed) // If we're closing and not going to hit within the next update, then wait.
+                                            var distance = (partHit.transform.position - vessel.CoM).sqrMagnitude;
+                                            var predictedDistance = (AIUtils.PredictPosition(partHit.transform.position, partHit.vessel.Velocity(), partHit.vessel.acceleration, Time.deltaTime) - AIUtils.PredictPosition(vessel, Time.deltaTime)).sqrMagnitude;
+                                            if (distance > predictedDistance && distance > Time.fixedDeltaTime*Time.fixedDeltaTime * (float)vessel.srfSpeed*(float)vessel.srfSpeed) // If we're closing and not going to hit within the next update, then wait.
```

Comment:
For clarity, add `Sqr` to the variable names. Also, it's better to write the condition as
```csharp
float threshold = Time.fixedDeltaTime * (float)vessel.srfSpeed;
if (distanceSqr > predictedDistanceSqr && distanceSqr > threshold * threshold)
```
if the value to be squared isn't already a single variable.

Reply:
Fixed this by using `missileDistancePerFrame.sqrtMagnitude` instead. I don't *think* there should be any issue with using that looking at the code, if anybody sees why surfSpeed would be needed instead of velocity (which I think should be surfSpeed in vector form when sub-orbital and in the atmosphere correct?) then please do speak up.

Reply:
Yeah, though `relativeSpeed` is badly named, since it's actually a distance, i.e., the change in the relative separation over the next frame. Feel free to rename it.

Reply:
I mean, it's technically still a speed, just in m/frame rather than m/s

Reply:
Only in the tautological sense that a distance moved in a frame is the velocity measured in m/frame at that moment.
But I guess `predictedChangeInSeparationOverTheNextFrame` is a bit cumbersome... so nevermind.